backend/fms_core/template_importer/row_handlers/library_capture/library_capture.py

```diff
@@ -0,0 +1,89 @@
+from fms_core.template_importer.row_handlers._generic import GenericRowHandler
+
+from fms_core.services.sample import get_sample_from_container
+from fms_core.services.container import get_container, get_or_create_container
+from fms_core.services.index import get_index
+from fms_core.services.library import capture_library
+
+
+
+class LibraryRowHandler(GenericRowHandler):
+    def __init__(self):
+        super().__init__()
+
+    def process_row_inner(self, capture_batch_info, source_sample, volume_used,
+                          comment, container, volume):
+
+        if not capture_batch_info:
+            self.errors['library_capture'] = 'No batch is associated with this capture.'
+
+        # Calling the service creator for Samples in LibraryPreparation
+        source_sample_obj, self.errors['container'], self.warnings['container'] = \
+            get_sample_from_container(barcode=source_sample['barcode'], coordinates=source_sample['coordinates'])
+
+        if not volume_used:
+            self.errors['volume_used'] = f"Volume used must be entered"
+        elif source_sample_obj and volume_used > source_sample_obj.volume:
```

Comment:
Accessing `source_sample.obj.volume` before validating if the source sample is not None will result in problems.

Reply:
isn't that exactly what is being done here ? if source_sample_obj is None python will never evaluate source_sample_obj.volume.

Reply:
Yeap, sorry. misread it